curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/6554a88d5af937226f4a9121.md

```diff
@@ -0,0 +1,71 @@
+---
+id: 6554a88d5af937226f4a9121
+title: Step 60
+challengeType: 20
+dashedName: step-60
+---
+
+# --description--
+
+The `.index()` method is identical to the `.find()` method but it throws a `ValueError` exception if it is unable to find the substring.
+
+Declare a variable called `offset` and give it the value of the index that `key_char` has in `alphabet`. Use `.index()` to find it.
+
+# --hints--
+
+You should declare a variable called `offset`.
+
+```js
+({ 
+  test: () => {
+    const vigenere = __helpers.python.getDef(code, "vigenere");
+    const {function_body} = vigenere;    
+    assert(function_body.match(/offset\s*=/));
+  }
+})
+```
+
+Your `offset` variable should store the value of `alphabet.index(key_char)`.
+
+```js
+({ 
+  test: () => {
+    const vigenere = __helpers.python.getDef(code, "vigenere");
+    const {function_body} = vigenere;    
+    assert(function_body.match(/offset\s*=\s*alphabet\.index\s*\(\s*key_char\s*\)/));
+  }
+})
+```
+
+# --seed--
+
+## --seed-contents--
+
+```py
+
+text = 'Hello Zaira'
+custom_key = 'python'
+
+def vigenere(message, key):
+    key_index = 0
+    alphabet = 'abcdefghijklmnopqrstuvwxyz'
+    encrypted_text = ''
+
+    for char in message.lower():
+    
+        # Append space to the message
+        if char == ' ':
+            encrypted_text += char
+        else:        
+            # Find the right key character to encode
+            key_char = key[key_index % len(key)]
+            key_index += 1
+--fcc-editable-region--
+            
+            index = alphabet.find(char)    
+            new_index = (index + offset) % len(alphabet)
+            encrypted_text += alphabet[new_index]
+    print('plain text:', text)
+    print('encrypted text:', encrypted_text)
+--fcc-editable-region--
```

Comment:
With the editable region here, and without explicit mention in the instructions, it seems like `offset = ` should be defined after the `print`s.